File: playwright-java/src/test/java/testtracks/data/TestData.java
package testtracks.data;

import java.security.SecureRandom;
import java.util.UUID;

public final class TestData {
  private static final SecureRandom RANDOM = new SecureRandom();

  private TestData() {}

  public static String newScenarioId() {
    return UUID.randomUUID().toString().replace("-", "").substring(0, 8);
  }

  public static int numericSuffix(int minimum, int maximumExclusive) {
    return RANDOM.nextInt(minimum, maximumExclusive);
  }

  public static String safeName(String prefix, String scenarioId, int maxLength) {
    var value = prefix + "_" + scenarioId;
    return value.length() <= maxLength ? value : value.substring(0, maxLength);
  }
}

File: playwright-csharp/tests/Data/TestData.cs
using System.Security.Cryptography;

namespace TestTracks.Playwright.CSharp.Specs.Data;

public static class TestData
{
    public static string NewScenarioId()
    {
        return Guid.NewGuid().ToString("N")[..8];
    }

    public static int NumericSuffix(int minimum = 7000, int maximumExclusive = 9999)
    {
        return RandomNumberGenerator.GetInt32(minimum, maximumExclusive);
    }

    public static string SafeName(string prefix, string scenarioId, int maxLength)
    {
        var value = $"{prefix}_{scenarioId}";
        return value.Length <= maxLength ? value : value[..maxLength];
    }
}
